docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md

```diff
@@ -0,0 +1,43 @@
+# This document lists known breaking changes in Roslyn after .NET 9 all the way to .NET 10.
```

Comment:
I always get confused on this so feel free to correct me.
Since the breaking changes listed below are shipping as part of .NET 9, I would expect them to be in the exist document that covers the `].NET 8, .NET 9]` window.

Reply:
You only see them when you turn on C# 14, which will be part of .NET 10.

Reply:
Ah, missed that. Thanks